### ISSUE TITLE
Conv2D accepts strings as 'filters' parameter, but fails to handle them correctly

### ISSUE THREAD

**Issue by bakaschwarz** [tensorflow/tensorflow#35190]
**System information**
- Have I written custom code: Yes
- OS Platform and Distribution: Linux, Linux-5.4.2-1-MANJARO-x86_64-with-arch-Manjaro-Linux
- TensorFlow installed from (source or binary): `tensorflow` binary using pip 
- TensorFlow version: 2.0.0
- Python version: 3.7.5
- CUDA/cuDNN version: -
- GPU model and memory: -

**Describe the current behavior**

I create a Conv2D layer with string `"12"` as parameter for `filters`.
Creation of the object does not fail. However trying to create a tensor out of
this object results in the following exception:
```
Traceback (most recent call last):
  File "/home/baka/.virtualenvs/cplace-machine-learning-backend/lib/python3.7/site-packages/IPython/core/interactiveshell.py", line 3291, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-26-60e14eee1394>", line 1, in <module>
    a(b)
  File "/home/baka/.virtualenvs/cplace-machine-learning-backend/lib/python3.7/site-packages/tensorflow_core/python/keras/engine/base_layer.py", line 817, in __call__
    self._maybe_build(inputs)
  File "/home/baka/.virtualenvs/cplace-machine-learning-backend/lib/python3.7/site-packages/tensorflow_core/python/keras/engine/base_layer.py", line 2141, in _maybe_build
    self.build(input_shapes)
  File "/home/baka/.virtualenvs/cplace-machine-learning-backend/lib/python3.7/site-packages/tensorflow_core/python/keras/layers/convolutional.py", line 165, in build
    dtype=self.dtype)
  File "/home/baka/.virtualenvs/cplace-machine-learning-backend/lib/python3.7/site-packages/tensorflow_core/python/keras/engine/base_layer.py", line 522, in add_weight
    aggregation=aggregation)
  File "/home/baka/.virtualenvs/cplace-machine-learning-backend/lib/python3.7/site-packages/tensorflow_core/python/training/tracking/base.py", line 744, in _add_variable_with_custom_getter
    **kwargs_for_getter)
  File "/home/baka/.virtualenvs/cplace-machine-learning-backend/lib/python3.7/site-packages/tensorflow_core/python/keras/engine/base_layer_utils.py", line 139, in make_variable
    shape=variable_shape if variable_shape else None)
  File "/home/baka/.virtualenvs/cplace-machine-learning-backend/lib/python3.7/site-packages/tensorflow_core/python/ops/variables.py", line 258, in __call__
    return cls._variable_v1_call(*args, **kwargs)
  File "/home/baka/.virtualenvs/cplace-machine-learning-backend/lib/python3.7/site-packages/tensorflow_core/python/ops/variables.py", line 219, in _variable_v1_call
    shape=shape)
  File "/home/baka/.virtualenvs/cplace-machine-learning-backend/lib/python3.7/site-packages/tensorflow_core/python/ops/variables.py", line 197, in <lambda>
    previous_getter = lambda **kwargs: default_variable_creator(None, **kwargs)
  File "/home/baka/.virtualenvs/cplace-machine-learning-backend/lib/python3.7/site-packages/tensorflow_core/python/ops/variable_scope.py", line 2507, in default_variable_creator
    shape=shape)
  File "/home/baka/.virtualenvs/cplace-machine-learning-backend/lib/python3.7/site-packages/tensorflow_core/python/ops/variables.py", line 262, in __call__
    return super(VariableMetaclass, cls).__call__(*args, **kwargs)
  File "/home/baka/.virtualenvs/cplace-machine-learning-backend/lib/python3.7/site-packages/tensorflow_core/python/ops/resource_variable_ops.py", line 1406, in __init__
    distribute_strategy=distribute_strategy)
  File "/home/baka/.virtualenvs/cplace-machine-learning-backend/lib/python3.7/site-packages/tensorflow_core/python/ops/resource_variable_ops.py", line 1537, in _init_from_args
    initial_value() if init_from_fn else initial_value,
  File "/home/baka/.virtualenvs/cplace-machine-learning-backend/lib/python3.7/site-packages/tensorflow_core/python/keras/engine/base_layer_utils.py", line 119, in <lambda>
    init_val = lambda: initializer(shape, dtype=dtype)
  File "/home/baka/.virtualenvs/cplace-machine-learning-backend/lib/python3.7/site-packages/tensorflow_core/python/ops/init_ops_v2.py", line 421, in __call__
    fan_in, fan_out = _compute_fans(scale_shape)
  File "/home/baka/.virtualenvs/cplace-machine-learning-backend/lib/python3.7/site-packages/tensorflow_core/python/ops/init_ops_v2.py", line 749, in _compute_fans
    fan_out = shape[-1] * receptive_field_size
TypeError: can't multiply sequence by non-int of type 'float'
```

You can also create a Conv2D object with other strings like `"test"` and it will not fail either.
Using "`test`" yields:
```
Traceback (most recent call last):
  File "/home/baka/.virtualenvs/cplace-machine-learning-backend/lib/python3.7/site-packages/IPython/core/interactiveshell.py", line 3291, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-35-60e14eee1394>", line 1, in <module>
    a(b)
  File "/home/baka/.virtualenvs/cplace-machine-learning-backend/lib/python3.7/site-packages/tensorflow_core/python/keras/engine/base_layer.py", line 817, in __call__
    self._maybe_build(inputs)
  File "/home/baka/.virtualenvs/cplace-machine-learning-backend/lib/python3.7/site-packages/tensorflow_core/python/keras/engine/base_layer.py", line 2141, in _maybe_build
    self.build(input_shapes)
  File "/home/baka/.virtualenvs/cplace-machine-learning-backend/lib/python3.7/site-packages/tensorflow_core/python/keras/layers/convolutional.py", line 165, in build
    dtype=self.dtype)
  File "/home/baka/.virtualenvs/cplace-machine-learning-backend/lib/python3.7/site-packages/tensorflow_core/python/keras/engine/base_layer.py", line 522, in add_weight
    aggregation=aggregation)
  File "/home/baka/.virtualenvs/cplace-machine-learning-backend/lib/python3.7/site-packages/tensorflow_core/python/training/tracking/base.py", line 744, in _add_variable_with_custom_getter
    **kwargs_for_getter)
  File "/home/baka/.virtualenvs/cplace-machine-learning-backend/lib/python3.7/site-packages/tensorflow_core/python/keras/engine/base_layer_utils.py", line 126, in make_variable
    variable_shape = tensor_shape.TensorShape(shape)
  File "/home/baka/.virtualenvs/cplace-machine-learning-backend/lib/python3.7/site-packages/tensorflow_core/python/framework/tensor_shape.py", line 776, in __init__
    self._dims = [as_dimension(d) for d in dims_iter]
  File "/home/baka/.virtualenvs/cplace-machine-learning-backend/lib/python3.7/site-packages/tensorflow_core/python/framework/tensor_shape.py", line 776, in <listcomp>
    self._dims = [as_dimension(d) for d in dims_iter]
  File "/home/baka/.virtualenvs/cplace-machine-learning-backend/lib/python3.7/site-packages/tensorflow_core/python/framework/tensor_shape.py", line 718, in as_dimension
    return Dimension(value)
  File "/home/baka/.virtualenvs/cplace-machine-learning-backend/lib/python3.7/site-packages/tensorflow_core/python/framework/tensor_shape.py", line 193, in __init__
    self._value = int(value)
ValueError: invalid literal for int() with base 10: 'test'
```

**Describe the expected behavior**

I expect that creation of the Conv2D object fails with a ValueError when used with a string or that it tries to convert the string into int.
If the string cannot be converted to int, an exception should be raised at the creation time of the layer, not when building the tensor. Right now, this is different behavior to parameters like `strides` which fails in the stage of building the layer object when presented with an not int-convertible string.

**Code to reproduce the issue**
```
import tensorflow.python.keras as k
a = k.layers.Conv2D(filters="12", kernel_size=3)
b = k.layers.Input((100, 100, 3))
a(b)
```
and
```
import tensorflow.python.keras as k
a = k.layers.Conv2D(filters="test", kernel_size=3)
b = k.layers.Input((100, 100, 3))
a(b)
```

**Comment by ravikyram** [comment]
@bakaschwarz 
Filters should be int but not strings. If we give integer value for filter we will not see any error message.
Please go through this [link](https://www.tensorflow.org/api_docs/python/tf/keras/layers/Conv2D?version=stable#arguments_2).Thanks!

**Comment by bakaschwarz** [comment]
> @bakaschwarz
> Filters should be int but not strings. If we give integer value for filter we will not see any error message.
> Please go through this [link](https://www.tensorflow.org/api_docs/python/tf/keras/layers/Conv2D?version=stable#arguments_2).Thanks!

You are totally right, that filters should be int, but not strings. That is exactly why i am reporting this though.
Because this gets accepted and can't be compiled into a tensor:
`a = k.layers.Conv2D(filters="test", kernel_size=3)`
This fails at layer creation time:
`a = k.layers.Conv2D(filters=12, kernel_size=3, strides="test")`
This gets accepted and creates a layer and can even be compiled into a tensor:
`a = k.layers.Conv2D(filters=12, kernel_size=3, strides="3")`

This is just inconsistent behavior. Why can i use strings for `strides` but not for `filters`. It should either work always or should always fail (which i think would be the correct behavior).

**Comment by jvishnuvardhan** [comment]
@bakaschwarz I agree, it is not consistent but it is not a bug as the `inputs` of different functions are clearly defined. I am not sure how much work required to make it kind of consistent.  Thanks!

**Comment by ymodak** [comment]
This is fixed latest tf-nightly version '2.2.0-dev20200228' . Thanks!
